src/gluonts/model/evaluation.py

```diff
@@ -0,0 +1,180 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+import logging
+from collections import ChainMap
+from typing import Iterable, Optional, Union
+from dataclasses import dataclass
+from toolz import first
+
+import numpy as np
+import pandas as pd
+
+from gluonts.dataset import DataEntry
+from gluonts.dataset.split import TestData
+from gluonts.time_feature.seasonality import get_seasonality
+from gluonts.model.forecast import Forecast
+from gluonts.ev import seasonal_error
+from gluonts.itertools import prod
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class BatchForecast:
+    """
+    Wrapper around ``Forecast`` objects, that adds a batch dimension
+    to arrays returned by ``__getitem__``, for compatibility with
+    ``gluonts.ev``.
+    """
+
+    forecast: Forecast
+    allow_nan: bool = False
+
+    def __getitem__(self, name):
+        value = self.forecast[name]
+        valid = ~np.isnan(value).any()
+
+        if not valid and self.allow_nan:
+            logger.warning(
+                "Forecast contains NaN values. Metrics may be incorrect."
+            )
+
+        if not valid and not self.allow_nan:
+            raise ValueError("Forecast contains NaN values")
+
+        return np.expand_dims(value, axis=0)
+
+
+def _get_data_batch(
+    input_: DataEntry,
+    label: DataEntry,
+    forecast: Forecast,
+    seasonality: Optional[int] = None,
+    mask_invalid_label: bool = True,
+    allow_nan_forecast: bool = False,
+) -> ChainMap:
```

Comment:
I don't think `ChainMap` is the interface that we want to expose. Can't this just be `dict`?